src/bridge/AdminRoomHandler.ts

```diff
@@ -35,6 +36,27 @@ enum CommandPermission {
     Admin,
 }
 
+/**
+ * For a given input event, try to parse out a command. This assumes
+ * the event may be malformed.
+ * @param event A matrix message event.
+ * @param prefix The prefix for the message.
+ * @returns A cmd (without prefix) and args
+ */
+export function parseCommandFromEvent(event: { content?: { body?: unknown }}, prefix = "!") {
+    if (typeof event.content?.body !== "string") {
+        return null;
+    }
+    // First line only
+    const body = event.content.body.trim().split("\n")[0];
+    if (!body.startsWith(prefix)) {
+        return null;
+    }
+    // Assumes all commands have the form "!wxyz [irc.server] [args...]"
+    const [cmd, ...args] = body.slice(prefix.length).split(" ");
```

Comment:
Should this check if `cmd` ends up being empty?  i.e. should `null` be returned for an `event.content.body` of `"!"`?

Reply:
Yes! Good point.